### PR TITLE
Update clap to version 4

### PR DESCRIPTION
**Pull request by msrd0**

### CODE REVIEW

**Comment by LukasKalbertodt** [comment]
Thanks for the PR! 
Unfortunately, this requires a major version bump as clap is a public dependency of this crate and since clap would raise the MSRV to 1.60. But I guess that's fine. 

Can you update the readme and the github actions to bump the MSRV to 1.60? And also, please use 4 spaces instead of tab to indent your changes (style CI is red).

I will review and test properly at a later time.

**Comment by msrd0** [comment]
> Can you update the readme and the github actions to bump the MSRV to 1.60? And also, please use 4 spaces instead of tab to indent your changes (style CI is red).

@LukasKalbertodt done

**Comment by LukasKalbertodt** [comment]
Wow, that was fast!
Please remove your "format code" commit. That's a ton of unrelated diff that doesn't really belong in this PR and should be done separately. 

I will only review in the coming days, when I have a proper setup to test this.

**Comment by msrd0** [comment]
@LukasKalbertodt Oh yeah that's annoying. I replaced the tabs with sed instead of rustfmt.

**Review by LukasKalbertodt — APPROVED** [review]
Looks good! Thank you

**Comment by LukasKalbertodt** [comment]
Released as v0.6.0

File: src/args.rs [hunk 1]
```diff
@@ -60,46 +60,46 @@ pub struct Arguments {
...
-    #[clap(
-        long = "--skip",
+    #[arg(
+        long = "skip",
         value_name = "FILTER",
-        number_of_values = 1,
+        num_args = 1,
         help = "Skip tests whose names contain FILTER (this flag can be used multiple times)",
     )]
     pub skip: Vec<String>,
```
Comment (epage) [inline src/args.rs:97]: FYI `num_args = 1` is likely not needed.  Most likely `number_of_values = 1` was being set back from the structopt days when a `Vec` implied `multiple = true`.  clap v3 split `multiple` into `multiple_values` and `multiple_occurrences` and `Vec` only implies `multiple_occurrences`.

Comment (LukasKalbertodt) [reply src/args.rs:97]: Thanks!

**Pull request merged**